package.json

```diff
@@ -51,14 +51,15 @@
   },
   "scripts": {
     "postinstall": "yarn --cwd ./examples",
-    "prebuild": "rimraf dist && npm run typegen",
-    "build": "rollup -c && yarn copy && rimraf dist/__tests__",
+    "build": "rimraf dist && npm run typegen && rollup -c && yarn copy && rimraf dist/__tests__ && npm run replace-reconciler-imports",
     "prepare": "npm run build",
     "eslint": "eslint --fix src/**/*.{ts,tsx}",
     "dev": "yarn --cwd ./examples start",
-    "test": "jest",
-    "typecheck": "tsc --noEmit --emitDeclarationOnly false --strict --jsx react",
-    "typegen": "tsc && mv dist/src/* dist && rm -rf dist/src || true",
+    "tests": "jest",
+    "test": "npm run typecheck && npm run eslint && npm run tests && npm run build",
+    "typecheck": "tsc --noEmit",
+    "replace-reconciler-imports": "npx replace 'import Reconciler' 'import * as Reconciler' dist -r --include='*.d.ts'",
```

Comment:
Sorry for ignorance, can you explain why we need to do this replace?

Reply:
To summarise all the ts-jest and replace and stuff changes:

People with `esModuleInterop` disabled get type errors when using default imports, which is why we do the replace, but ES modules/babel/rollup does not support the calling TypeScript namespaces (`import * as`) like functions (a feature of TypeScript) so we have to use a default import in this project in order to export as an ES module. This works for ES modules, and when using babel (which is why I've swapped ts-jest for babel - closer mirrors the build pipeline).

Probably worth having a read of the following for context: #827 #829

Reply:
Thank you for the pointers! I always had esModuleInterop enabled so I never saw any problem.